contrib/babelfishpg_tsql/sql/babelfishpg_tsql.sql

```diff
@@ -1031,30 +1029,42 @@ CREATE OR REPLACE FUNCTION sys.sp_tables_internal(
 			CAST(table_type AS sys.varchar(32)) AS TABLE_TYPE,
 			CAST(remarks AS sys.varchar(254)) AS REMARKS
 			FROM sys.sp_tables_view
-			WHERE ((SELECT coalesce(in_table_name,'')) = '' OR table_name = in_table_name collate sys.database_default)
-			AND ((SELECT coalesce(in_table_owner,'')) = '' OR table_owner = in_table_owner collate sys.database_default)
-			AND ((SELECT coalesce(in_table_qualifier,'')) = '' OR table_qualifier = in_table_qualifier collate sys.database_default)
-			AND ((SELECT coalesce(cs_as_in_table_type,'')) = ''
-			    OR table_type = opt_table
-			    OR table_type = opt_view)
+			WHERE (in_table_name IS NULL OR table_name = in_table_name collate sys.database_default)
+			AND (in_table_owner IS NULL OR table_owner = in_table_owner collate sys.database_default)
+			AND (in_table_qualifier IS NULL OR table_qualifier = in_table_qualifier collate sys.database_default)
+			AND (cs_as_in_table_type IS NULL OR table_type = opt_table OR table_type = opt_view)
 			ORDER BY table_qualifier, table_owner, table_name;
 		END IF;
 	END;
 $$
 LANGUAGE plpgsql STABLE;
-	 
 
 CREATE OR REPLACE PROCEDURE sys.sp_tables (
-    "@table_name" sys.nvarchar(384) = '',
-    "@table_owner" sys.nvarchar(384) = '', 
-    "@table_qualifier" sys.sysname = '',
-    "@table_type" sys.nvarchar(100) = '',
+    "@table_name" sys.nvarchar(384) = NULL,
+    "@table_owner" sys.nvarchar(384) = NULL, 
+    "@table_qualifier" sys.sysname = NULL,
+    "@table_type" sys.nvarchar(100) = NULL,
     "@fusepattern" sys.bit = '1')
 AS $$
 	DECLARE @opt_table sys.varchar(16) = '';
-	DECLARE @opt_view sys.varchar(16) = ''; 
+	DECLARE @opt_view sys.varchar(16) = '';
 BEGIN
-	IF (@table_qualifier != '') AND (LOWER(@table_qualifier) != LOWER(sys.db_name()))
+
+	-- Handle special case: Enumerate all databases when name and owner are blank but qualifier is '%'
+	IF (@table_qualifier = '%' AND @table_owner = '' AND @table_name = '')
```

Comment:
As we are checking whether table_owner or table_name is == '' or not, let say I EXEC this PROC by passing % as table_qualifier, will this condition satisfy? My understanding is NULL and '' are not same, based on that I am asking this.

Reply:
RIght, `NULL` and `''` are not same. If we simply pass `'%'` as table_qualifier, this condition won't be satisfied and an error will be thrown.

Reply:
@thephantomthief Yes